artifacts/definitions/Generic/Utils/SendEmail.yaml

```diff
@@ -9,62 +9,117 @@ parameters:
   default: gmail
   description: The name of the secret to use to send the mail with.
 
-- name: Receipient
-  default: noone@gmail.com
+- name: Recipients
+  type: json_array
+  default: '["noone@example.org"]'
   description: Where to send the mail to.
 
+- name: Sender
+  description: The sender address (from).
+
 - name: FilesToUpload
   type: json_array
   default: '["C:/test.txt"]'
 
 - name: PlainTextMessage
-  default: A test email
+  description: A plain-text message.
+
+- name: HTMLMessage
+  description: An HTML-formatted message.
 
 - name: Subject
   default: A message from Velociraptor
 
-export: |
-   LET Boundary = format(format="-----------------------------9051914041544843%v",
-                         args=now())
-
-   -- A Helper function to make a plain text message.
-   LET PlainData(Value) = format(
-       format='--%s\r\nContent-Type: text/plain; charset="utf-8"\r\n\r\n%v\r\n',
-       args=[Boundary, Value])
-
-   -- Encodes the file as base64 with lines split on 80 chars
-   LET EncodeFile(Filename) = regex_replace(
-       re="(.{80})",
-       replace="$1\r\n",
-       source=base64encode(string=read_file(filename=Filename)))
-
-   -- A Helper function to embed a file content from disk.
-   LET AttachFile(Filename) = format(
-       format='--%s\r\nContent-Type: application/octet-stream; name="%s"\r\nContent-Disposition: attachment; filename="%s"\r\nContent-Transfer-Encoding: base64\r\n\r\n%v\r\n\r\n',
-       args=[Boundary, basename(path=Filename),
-             basename(path=Filename), EncodeFile(Filename=Filename)])
-
-   -- The End boundary signals the last part
-   LET END = format(format="%s--\r\n", args=Boundary)
+- name: Period
+  type: int
+  default: 10
+  description: |
+    Refuse to send mails more often than this interval (in seconds). This throttling
+    is applied to the whole server.
 
 sources:
 - query: |
+    -- Create a random string suitable as a MIME boundary:
```

Comment:
Please put the main logic back in the export section so it can be used by other artifacts.